operator/v1/types_ingress.go

```diff
@@ -282,6 +287,339 @@ var (
+	//
+	// and looks like this (yaml):
+	//
+	//   ciphers:
+	//     - ECDHE-ECDSA-AES256-GCM-SHA384
+	//     - ECDHE-RSA-AES256-GCM-SHA384
+	//     - ECDHE-ECDSA-CHACHA20-POLY1305
+	//     - ECDHE-RSA-CHACHA20-POLY1305
+	//     - ECDHE-ECDSA-AES128-GCM-SHA256
+	//     - ECDHE-RSA-AES128-GCM-SHA256
+	//     - ECDHE-ECDSA-AES256-SHA384
+	//     - ECDHE-RSA-AES256-SHA384
+	//     - ECDHE-ECDSA-AES128-SHA256
+	//     - ECDHE-RSA-AES128-SHA256
+	//   securityProtocol:
+	//     minimumVersion: TLSv1.2
+	//     maximumVersion: TLSv1.2
+	//   dhParamSize: 2048
+	//
+	// +optional
+	// +nullable
+	Modern *ModernSecurityProfile `json:"modern,omitempty"`
+	// custom is a user-defined security profile. An example custom profile
+	// looks like this:
+	//
+	//   ciphers:
+	//     - ECDHE-ECDSA-CHACHA20-POLY1305
+	//     - ECDHE-RSA-CHACHA20-POLY1305
+	//     - ECDHE-RSA-AES128-GCM-SHA256
+	//     - ECDHE-ECDSA-AES128-GCM-SHA256
+	//   securityProtocol:
+	//     minimumVersion: TLSv1.1
+	//     maximumVersion: TLSv1.2
+	//   dhParamSize: 1024
+	//
+	// Any unset fields of a custom security profile default to the IntermediateSecurityProfile.
+	// 
+	// +optional
+	// +nullable
+	Custom *CustomSecurityProfile `json:"custom,omitempty"`
+}
+
+type OldSecurityProfile struct{}
+type IntermediateSecurityProfile struct{}
+type ModernSecurityProfile struct{}
+
+// CustomSecurityProfile defines the schema for a custom security profile.
+type CustomSecurityProfile struct {
+	SecurityProfile `json:"securityProfile"`
+}
+
+// SecurityProfileType defines a type of security profile.
+type SecurityProfileType string
+
+const (
+	// Old is a security profile based on:
+	// https://wiki.mozilla.org/Security/Server_Side_TLS#Old_backward_compatibility
+	SecurityProfileOldType SecurityProfileType = "Old"
+	// Intermediate is a security profile based on:
+	// https://wiki.mozilla.org/Security/Server_Side_TLS#Intermediate_compatibility_.28default.29
+	SecurityProfileIntermediateType SecurityProfileType = "Intermediate"
+	// Modern is a security profile based on:
+	// https://wiki.mozilla.org/Security/Server_Side_TLS#Modern_compatibility
+	SecurityProfileModernType SecurityProfileType = "Modern"
+	// Custom is a security profile that allows for user-defined parameters.
+	SecurityProfileCustomType SecurityProfileType = "Custom"
+)
+
+// SecurityProfile defines the schema for a security profile.
+type SecurityProfile struct {
+	// ciphers is used to specify the cipher algorithms that are negotiated
+	// during the SSL/TLS handshake with an IngressController. Preface a cipher
+	// with a "!" to disable a specific cipher from being negotiated. For example,
+	// to use DES-CBC3-SHA but not EDH-DSS-DES-CBC3-SHA (yaml):
+	//
+	//   ciphers:
+	//     - DES-CBC3-SHA
+	//     - !EDH-DSS-DES-CBC3-SHA
+	//
+	// If unset, Ciphersuites are determined by the SecurityProfileType.
+	//
+	// +optional
+	Ciphers []*string `json:"ciphers,omitempty"`
```

Comment:
This might be an error... did you mean a pointer to the array?

```go
Ciphers *[]string `json:"ciphers,omitempty"`
```

Reply:
I did, thanks for catching this.